core/controllers/blog_homepage.py

```diff
@@ -0,0 +1,170 @@
+        if not blog_post:
+            raise self.PageNotFoundException(
+                Exception(
+                    'The blog post page with the given url doesn\'t exist.'))
+        user_settings = user_services.get_user_settings(blog_post.author_id)
+        blog_post_dict = (
+            blog_services.get_blog_post_from_model(blog_post).to_dict())
+        blog_post_dict['author_name'] = user_settings.username
+        del blog_post_dict['author_id']
+        blog_post_summaries = (
+            blog_services.get_published_blog_post_summaries_by_user_id(
+                blog_post.author_id,
+                feconf.MAX_POSTS_TO_RECOMMEND_AT_END_OF_BLOG_POST))
+        blog_post_summary_dicts = (
+            _get_blog_card_summary_dicts_for_homepage(blog_post_summaries))
+
+        self.values.update({
+            'profile_picture_data_url': user_settings.profile_picture_data_url,
+            'blog_post_dict': blog_post_dict,
+            'summary_dicts': blog_post_summary_dicts
+        })
+        self.render_json(self.values)
+
+
+class AuthorsPageHandler(base.BaseHandler):
+    """Provides blog cards data and author data for the authors page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+    URL_PATH_ARGS_SCHEMAS = {
+        'author_username': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {},
+    }
+
+    @acl_decorators.open_access
+    def get(self, author_username):
+        """Handles GET requests."""
+        user_settings = (
+            user_services.get_user_settings_from_username(author_username))
+        if user_settings:
+            if user_settings.role != (BLOG_ADMIN or BLOG_POST_EDITOR):
+                raise self.PageNotFoundException(
+                    Exception(
+                        'The given user is not a blog post author.'))
+            blog_post_summaries = (
+                blog_services.get_published_blog_post_summaries_by_user_id(
+                    user_settings.user_id,
+                    feconf.MAX_LIMIT_FOR_CARDS_ON_BLOG_AUTHORS_PAGE))
+            if blog_post_summaries:
+                blog_post_summary_dicts = (
+                    _get_blog_card_summary_dicts_for_homepage(
+                        blog_post_summaries))
+            else:
+                blog_post_summary_dicts = None
+
+            self.values.update({
+                'author_name': author_username,
+                'profile_picture_data_url': (
+                    user_settings.profile_picture_data_url),
+                'author_bio': user_settings.user_bio,
+                'summary_dicts': blog_post_summary_dicts
+            })
+            self.render_json(self.values)
+
+        else:
+            raise self.PageNotFoundException(
+                Exception(
+                    'User with given username does not exist'))
```

Comment:
Try to return asap in the function definition, this will reduce branching + makes the code more readable.
```
if user_settings is None:
    raise self.PageNotFoundException(...)

if user_settings.role != (BLOG_ADMIN or BLOG_POST_EDITOR):
     raise self.PageNotFoundException(...)

blog_post_summaries = ...
..
..
```

Reply:
Done.